### ISSUE TITLE
In a migration, how to refer to the type created by db.schema.createType

### ISSUE THREAD

**Issue by jroitgrund** [kysely-org/kysely#1269]
When using a custom schema

```
export default defineConfig({
  dialect: new PostgresDialect({
    pool: new Pool({
      connectionString: process.env.DATABASE_URL,
    }),
  }),
  plugins: [new WithSchemaPlugin('custom-schema')],
  migrations: {
    migrationTableSchema: 'custom-schema',
    migrationFolder: 'src/db/migrations',
  },
})
```

the following migraiton fails:

```
  await db.schema
    .createType('enum_type')
    .asEnum([
      'value'
    ])
    .execute()
  await db.schema
    .createTable('projects')
    .addColumn('enum', sql`enum_type`))
    .execute()
```

I get 

```
    error: type "enum_type" does not exist
```

(presumably because the type needs to be prefixed with the schema).

How can I fix this? I don't know how to access the schema name from the migration to manually prefix it.

Note that none of the sql helpers (.ref, .table, etc) seem to fix this.

**Comment by jroitgrund** [comment]
It looks like `(db as any).schema.executor.plugins[0].transformer.schema` exposes the schema but the private properties are only accessible in a debugger.

**Comment by yrahcaz** [comment]
Try wrapping the type in double quotes:

`.addColumn('enum', sql`"enum_type"`))`

**Comment by jroitgrund** [comment]
I get the same error with double quotes

**Comment by jroitgrund** [comment]
Sorry - I wrote but didn't run the minimal repro above, hence the syntax error.

I've now tried running the actual minimal repro as the only migration on a fresh DB:

```
import { PostgresDialect, WithSchemaPlugin } from 'kysely'
import { defineConfig } from 'kysely-ctl'
import { Pool } from 'pg'

export default defineConfig({
  dialect: new PostgresDialect({
    pool: new Pool({
      connectionString: process.env.DATABASE_URL,
    }),
  }),
  plugins: [new WithSchemaPlugin('other')],
  migrations: {
    migrationTableSchema: 'other',
    migrationFolder: 'src/db/migrations',
  },
})
```

```
import { type Kysely, sql } from 'kysely'

export async function up(db: Kysely<any>): Promise<void> {
  await db.schema.createType('enum_type').asEnum(['value']).execute()
  await db.schema
    .createTable('projects')
    .addColumn('enum', sql`enum_type`)
    .execute()
}

export async function down(db: Kysely<any>): Promise<void> {}
```

(this is the only migration in src/db/migrations and the DB is brand new

```
$ DATABASE_URL=postgresql://postgres:postgres@localhost:5432/db pnpm kysely migrate:latest
◐ Starting migration to latest
✖ Migration failed with error: type "enum_type" does not exist @ "1732135764590_project"
```

This happens both with and without double quotes.

If I remove the createTable statement, the type is correctly created in the target schema:

```
db=# SET search_path TO other;
SET
db=# \dT
        List of data types
 Schema |   Name    | Description
--------+-----------+-------------
 other  | enum_type |
(1 row)
```

**Comment by koskimas** [comment]
Kysely doesn't parse raw SQL and therefore has no idea what you wrote inside your raw sql snippet. For that reason, it can't prefix it with the schema.

We already have the `sql.id` function that can be used to create raw identifiers in a way kysely knows it's an identifier (table name, column name, type name etc.). But we don't prefix those either unfortunately.

We should add something like `sql.schemableId` to be able to handle these cases. It's definitely not a clean solution.

Alternatively we could allow ANY string in the second argument of `addColumn` as I think @igalklebanov suggested somewhere. Then we could treat any type as a "schemable id" automatically.

**Comment by jroitgrund** [comment]
Thanks for taking a look! Do you have a suggested workaround in the meantime? I tried setting the schema as a field on the DB instance before migrating so I can manually prepend it to the type name, but since the DB object passed to the migrator is actually a transaction, this doesn't work.

>Then we could treat any type as a "schemable id" automatically.

My hunch is that schemableId would work better than implicitly treating anything 'unrecognized' as schemable, because some of the types which currently require raw SQL aren't "schemable" (e.g. types added by extensions, array types).

For example,

```
.addColumn('list, sql.raw(`${schema}.integer[]`), (col) => col.notNull())

✖ Migration failed with error: type "other.integer[]" does not exist @ "1731672569480_project"                                                                  
```

**Comment by koskimas** [comment]
Yeah there's a way. I can write a https://kyse.link for you in a bit.

**Comment by koskimas** [comment]
Actually I couldn't figure out a way to do that currently 😞

I think the only way is to somehow pass in the schema using an environment variable or a global variable and glue it to the type manually 😬

**Comment by jroitgrund** [comment]
Unfortunately, my use case is one-schema-per-test and running tests concurrently. I'll keep an eye on the issue, thanks for the quick replies :)

**Comment by matthew-holder-revvity** [comment]
That is actually an issue in many situations where you need to use references and IDs, but no way of knowing what, if any, schema should be prefixed to them. There also isn't really any reliable way right now to get the schema from the connection without using hacky methods.

**Comment by Ahmed1813** [comment]
I think it shouldn't be an issue if we can get schema directly from the connection, something like 
```
const tenantDb = db.withSchema("tenant")
console.log(tenantDb.currentSchema)
// or
console.log(tenantDb.schema) 
```
then we can manually add schema.
